grove/tomography/utils.py

```diff
@@ -0,0 +1,627 @@
+            raise ValueError("transfer_matrix() only implemented for orthonormal operator bases")
+        return self.basis_transform.H * sop.data * self.basis_transform
+
+    def super_from_tm(self, tm):
+        """
+        Reconstruct a super operator from a transfer matrix representation.
+        This inverts `self.transfer_matrix(...)`.
+
+        :param (numpy.ndarray) tm: A process in transfer matrix form.
+        :return: A qutip super operator.
+        :rtype: qutip.Qobj.
+        """
+        if not self.is_orthonormal():  # pragma no coverage
+            raise ValueError("super_from_tm() only implemented for orthonormal operator bases")
+
+        data = self.basis_transform * tm * self.basis_transform.H
+        sop = qt.Qobj(data, dims=[self.ops[0].dims, self.ops[0].dims])
+        sop.superrep = "super"
+        return sop
+
+    def __str__(self):
+        return "<span[{}]>".format(",".join(self.labels))
+
+    def __eq__(self, other):
+        return (self.labels == other.labels and all(
+            [(my_op - o_op).norm('fro') < EPS for (my_op, o_op) in zip(self.ops, other.ops)]))
+
+
+PAULI_BASIS = OperatorBasis(
+    [("I", qI / np.sqrt(2)), ("X", qX / np.sqrt(2)),
+     ("Y", qY / np.sqrt(2)), ("Z", qZ / np.sqrt(2))])
+
+
+def gell_mann_basis(qudit_dim):
+    """
+    Compute an orthonormal hermitian operator basis for the space of observables of a qudit with
+    `d` levels. These are identical to the Gell-Mann matrices up to normalization.
+    See also this
+    `wikipedia article <https://en.wikipedia.org/wiki/Generalizations_of_Pauli_matrices>`_ .
+
+    :param int qudit_dim: The dimension of the underlying Hilbert space.
+    :return: The normalized Gell-Mann operator basis.
+    :rtype: OperatorBasis
+    """
+
+    labels_ops = [('I', qt.qeye(qudit_dim) / np.sqrt(qudit_dim))]
+    xops = []
+    yops = []
+
+    # off diagonals
+    for jj in range(qudit_dim):
+        for kk in range(jj):
+            sigma_jk = qt.projection(qudit_dim, jj, kk)
+            xops.append(('X{}{}'.format(kk, jj), (sigma_jk + sigma_jk.dag()) / np.sqrt(2)))
+            yops.append(('Y{}{}'.format(kk, jj), (sigma_jk - sigma_jk.dag()) / np.sqrt(2) * 1j))
+
+    labels_ops += xops + yops
+
+    # diagonals
+    for jj in range(1, qudit_dim):
+        diag = np.array([1.] * jj + [-jj] + [0.] * (qudit_dim - jj - 1))
+        diag /= np.linalg.norm(diag, 2)
+        labels_ops.append(('Z{}'.format(jj),
+                           qt.qdiags([diag], [0], [[qudit_dim], [qudit_dim]],
+                                     (qudit_dim, qudit_dim))))
+
+    return OperatorBasis(labels_ops)
+
+
+def n_qubit_pauli_basis(n):
+    """
+    Construct the tensor product operator basis of `n` PAULI_BASIS's.
+
+    :param int n: The number of qubits.
+    :return OperatorBasis: The product Pauli operator basis of `n` qubits
+    """
+    if n >= 1:
+        return PAULI_BASIS ** n
+    else:  # pragma no coverage
```

Comment:
We don't use pragma no coverage anywhere else... maybe we should just add tests for these.